### PR TITLE
Use more pflag functions instead of writing it ourselves.

### PR DESCRIPTION
**Pull request by eparis**
We implement our own mechanisms to munge around pflags and golang flags.  pflags now implements these.  Use pflags.

### CODE REVIEW

**Comment by eparis** [comment]
This should land after https://github.com/kubernetes/kubernetes/pull/12745  otherwise you get a warning about a flag with _ because thing changes when pflag.CommandLine is merged.

**Comment by thockin** [comment]
LGTM

**Comment by thockin** [comment]
@krousey to review after #12745

File: pkg/util/flags.go [hunk 1]
```diff
@@ -16,10 +16,13 @@ limitations under the License.
 
 package util
 
-import "strings"
+import (
+	goflag "flag"
```
Comment (krousey) [inline pkg/util/flags.go]: Why rename this?

**Comment by krousey** [comment]
This looks good. I will add the LGTM label once the other PR goes through.

**Comment by krousey** [comment]
You have some unit test failures in Shippable. You should be able to reproduce them with `hack/test-go.sh`

**Comment by krousey** [comment]
@k8s-bot test this

**Comment by krousey** [comment]
@k8s-bot test this again

**Comment by krousey** [comment]
@eparis This is consistently failing with

```console
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:206
Aug 18 13:01:35.947: Failed to parse kubectl port-forward output: W0818 13:01:35.947476    8628 flags.go:39] log_flush_frequency is DEPRECATED and will be removed in a future version. Use log-fl
```

Is this expect until the other one goes in?

**Comment by krousey** [comment]
@k8s-bot test this

**Comment by eparis** [comment]
yes, i would expect that failure until after the other PR

**Comment by krousey** [comment]
Waiting on #12745 to get merged, then I'll re-run the tests.

**Comment by krousey** [comment]
@eparis Mind if we try to get this rebased and in soon. I'm going on PTO after tomorrow morning. If not, I'll just re-assign this before I go.

**Comment by eparis** [comment]
rebased.

**Comment by saad-ali** [comment]
Another rebase needed :\

**Comment by eparis** [comment]
rebased again.

**Comment by saad-ali** [comment]
Thanks Eric. Will merge on green

**Comment by eparis** [comment]
green!

**Comment by saad-ali** [comment]
Merged :)

**Pull request merged**